inst/tests/tests.Rraw

```diff
@@ -14071,9 +14071,20 @@ dn = as.POSIXct('2016-09-18 08:00:00')
 up = as.POSIXct('2016-09-18 09:00:00')
 test(2028.11, between(x, dn, up, verbose=TRUE), output="between parallel processing of double using closed bounds with recycling took")
 test(2028.12, between(x, dn, up, incbounds=FALSE, verbose=TRUE), output="between parallel processing of double using open bounds with recycling took")
+
+## also handling of string lower/upper bounds _only when x has a time zone_
```

Comment:
Actually this should work for arbitrary class, not just character.

Reply:
My first concern is whether this will produce confusing warnings/errors or some classes, but I guess that's just a matter of getting an error at `x>=lower & x<=upper` in the non-parallel branch or getting it at `as.POSIXct`.

Second concern is for `inherits(lower 'POSIXct')` with mismatched timezones. I've always found it odd that `as.POSIXct(.POSIXct(0, tz = 'UTC'), tz = 'US/Arizona')` will ignore the new timezone. But, I guess that doesn't matter since it doesn't change the numeric representation anyway.

Reply:
Errors would be definitely confused, there was no tests for that so you haven't noticed. I added more robust coercion, and error handling. Timezone mismatch is also a little bit more robust now. Tests will follow, want to check coverage report first.